solana/programs/example-native-token-transfers/src/instructions/initialize.rs

```diff
@@ -37,12 +37,6 @@ pub struct Initialize<'info> {
     )]
     pub config: Box<Account<'info, crate::config::Config>>,
 
-    #[account(
-        constraint =
-            args.mode == Mode::Locking
-            || mint.mint_authority.unwrap() == token_authority.key()
-            @ NTTError::InvalidMintAuthority,
-    )]
     pub mint: Box<InterfaceAccount<'info, token_interface::Mint>>,
```

Comment:
maybe add a note here that the mint is unconstrained in the accounts struct and is the responsibility of the handler to constrain it